src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -0,0 +1,452 @@
+  rw degree_map_eq_of_leading_coeff_ne_zero (int.cast_ring_hom R) _,
+  { cases n with k,
+    { simp only [cyclotomic, degree_one, dif_pos, nat.totient_zero, with_top.coe_zero]},
+    rw [←degree_cyclotomic' (complex.is_primitive_root_exp k.succ (nat.succ_ne_zero k))],
+    exact (int_cyclotomic_spec k.succ).2.1 },
+  simp only [(int_cyclotomic_spec n).right.right, ring_hom.eq_int_cast, monic.leading_coeff,
+  int.cast_one, ne.def, not_false_iff, one_ne_zero]
+end
+
+/-- The natural degree of `cyclotomic n` is `totient n`. -/
+lemma nat_degree_cyclotomic (n : ℕ) (R : Type*) [ring R] [nontrivial R] :
+  (cyclotomic n R).nat_degree = nat.totient n :=
+begin
+  have hdeg := degree_cyclotomic n R,
+  rw degree_eq_nat_degree (cycl_ne_zero n R) at hdeg,
+  norm_cast at hdeg,
+  exact hdeg
+end
+
+/-- `∏ i in nat.divisors n, cyclotomic i R = X ^ n - 1`. -/
+lemma prod_cyclotomic_eq_X_pow_sub_one {n : ℕ} (hpos : 0 < n) (R : Type*) [comm_ring R] :
+  ∏ i in nat.divisors n, cyclotomic i R = X ^ n - 1 :=
+begin
+  have integer : ∏ i in nat.divisors n, cyclotomic i ℤ = X ^ n - 1,
+  { have mapinj : function.injective (map (int.cast_ring_hom ℂ)),
+    { apply map_injective,
+      simp only [int.cast_injective, int.coe_cast_ring_hom] },
+    apply mapinj,
+    rw map_prod (int.cast_ring_hom ℂ) (λ i, cyclotomic i ℤ),
+    simp only [int_cyclotomic_spec, map_pow, nat.cast_id, map_X, map_one, ring_hom.coe_of, map_sub],
+    exact prod_cyclotomic'_eq_X_pow_sub_one hpos
+    (complex.is_primitive_root_exp n (ne_of_lt hpos).symm) },
+  have coerc : X ^ n - 1 = map (int.cast_ring_hom R) (X ^ n - 1),
+  { simp only [map_pow, map_X, map_one, map_sub] },
+  have h : ∀ i ∈ n.divisors, cyclotomic i R = map (int.cast_ring_hom R) (cyclotomic i ℤ),
+  { intros i hi,
+    exact (map_cyclotomic_int i R).symm },
+  rw [finset.prod_congr (refl n.divisors) h, coerc, ←map_prod (int.cast_ring_hom R)
+  (λ i, cyclotomic i ℤ), integer]
+end
+
+/-- We have
+`cyclotomic n R = (X ^ k - 1) /ₘ (∏ i in nat.proper_divisors k, cyclotomic i K)`. -/
+lemma cyclotomic_eq_X_pow_sub_one_div {R : Type*} [comm_ring R] [nontrivial R] {n : ℕ}
+  (hpos: 0 < n) : cyclotomic n R = (X ^ n - 1) /ₘ (∏ i in nat.proper_divisors n, cyclotomic i R) :=
+begin
+  rw [←prod_cyclotomic_eq_X_pow_sub_one hpos,
+  nat.divisors_eq_proper_divisors_insert_self_of_pos hpos,
+  finset.prod_insert nat.proper_divisors.not_self_mem],
+  have prod_monic : (∏ i in nat.proper_divisors n, cyclotomic i R).monic,
+  { apply monic_prod_of_monic,
+    intros i hi,
+    exact cyclotomic.monic i R },
+  rw (div_mod_by_monic_unique (cyclotomic n R) 0 prod_monic _).1,
+  simp only [degree_zero, zero_add],
+  split,
+  { rw mul_comm },
+  rw [bot_lt_iff_ne_bot],
+  intro h,
+  exact monic.ne_zero prod_monic (degree_eq_bot.1 h)
+end
+
+/-- If there is a primitive `n`-th root of unity in `K`, then `cyclotomic n K = cyclotomic' n K`. -/
+lemma cyclotomic_eq_cyclotomic' {K : Type*} [field K] {ζ : K} {n : ℕ} (h : is_primitive_root ζ n) :
```

Comment:
Do you want to expose `cyclotomic'` or do you view it as implementation detail?
If it is an implementation detail, then I think you should restate this so that it doesn't mention `cyclotomic'`, but instead write the product over primitive roots directly.

Reply:
I just felt that it was natural to provides a link between the two definitions, but you're right, it is probably more interesting as a property of `cyclotomic n R`, without mentioning `cyclotomic'`.